validation/citrus-validation-xml/src/test/java/org/citrusframework/validation/xml/schema/XmlSchemaValidationTest.java

```diff
@@ -0,0 +1,110 @@
+package org.citrusframework.validation.xml.schema;
+
+import org.citrusframework.context.TestContext;
+import org.citrusframework.context.TestContextFactory;
+import org.citrusframework.exceptions.CitrusRuntimeException;
+import org.citrusframework.message.DefaultMessage;
+import org.citrusframework.message.Message;
+import org.citrusframework.validation.xml.XmlMessageValidationContext;
+import org.citrusframework.xml.XsdSchemaRepository;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.xml.xsd.SimpleXsdSchema;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.util.Optional;
+
+import static org.citrusframework.validation.xml.schema.XmlSchemaValidation.FAIL_IF_NO_SCHEMA_FOUND_ENV_VAR_NAME;
+import static org.citrusframework.validation.xml.schema.XmlSchemaValidation.FAIL_IF_NO_SCHEMA_FOUND_PROPERTY_NAME;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.springframework.test.util.ReflectionTestUtils.getField;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+public class XmlSchemaValidationTest {
+
+    @DataProvider(name = "booleans")
+    public Object[] dataProvider() {
+        return new Object[] {
+                true,
+                false
```

Comment:
if this was booleans, I would always test "other values" (e.g. exactly this value) to see what the behavior would be. but since I would change this to an `enum`, it is another story entirely.